Source/WebCore/platform/graphics/avfoundation/objc/MediaPlayerPrivateMediaSourceAVFObjC.mm

```diff
@@ -271,19 +215,84 @@ void getSupportedTypes(HashSet<String>& types) const final
 {
     ALWAYS_LOG(LOGIDENTIFIER);
 
-    if (RefPtr mediaSourcePrivate = downcast<MediaSourcePrivateAVFObjC>(client.mediaSourcePrivate())) {
-        mediaSourcePrivate->setPlayer(this);
-        m_mediaSourcePrivate = WTFMove(mediaSourcePrivate);
-        client.reOpen();
-    } else
-        m_mediaSourcePrivate = MediaSourcePrivateAVFObjC::create(*this, client);
+    m_renderer->notifyWhenErrorOccurs([weakThis = WeakPtr { *this }](PlatformMediaError) {
```

Comment:
Can you add in the commit log why you are moving the callback setting from constructor to load method?